File: 2003/VB101SamplesAll/VB101SamplesBCL2/UsingNGEN/SharedLib1/Numbers.vb
Public Class Numbers
    Public Function CalculateMean(ByVal start As Long, ByVal count As Long, ByVal stepVal As Long) As Long
        Dim total As Long = start
        Dim mean As Long = 0

        Try
            Dim i As Long
            For i = start To count - 1 Step i + stepVal

                total += i
            Next

            mean = total / count
        Catch
            mean = -1
        End Try

        Return mean
    End Function
End Class

File: 2003/CS101SamplesAll/CS101SamplesBCL2/UsingNGEN/SharedLib1/Numbers.cs
using System;
using System.Collections.Generic;
using System.Text;


namespace SharedLib1
{
    public class Numbers
    {
        public long CalculateMean(long start, long count, long step)
        {
            long total = start;
            long mean = 0;

            try
            {
                for (long i = start; i < count; i += step)
                {
                    total += i;
                }

                mean = total / count;
            }
            catch (Exception)
            {
                mean = -1;
            }

            return mean;
        }
    }
}
